### PR TITLE
[GLUTEN-5341][VL] Fix SPARK-42782: Hive compatibility check for get_json_object

### PR DESCRIPTION
**Pull request by ayushi-agarwal**
## What changes were proposed in this pull request?

Fixes get_json_object test introduced in spark 3.5 to check compatibility with hive. * character is not supported in velox in json queries.

(Fixes: \#5341)

## How was this patch tested?
Ran in local

### CODE REVIEW

**Comment by ayushi-agarwal** [comment]
@gaoyangxiaozhu @yma11

**Comment by ayushi-agarwal** [comment]
@zhli1142015 Could you please review? Thanks

File: gluten-ut/spark35/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -126,7 +126,7 @@ class VeloxTestSettings extends BackendTestSettings {
   enableSuite[GlutenHashExpressionsSuite]
   enableSuite[GlutenIntervalExpressionsSuite]
   enableSuite[GlutenJsonFunctionsSuite]
-    // Disable for Spark3.5.
+    // * in get_json_object expression not supported in velox
```
Comment (zhli1142015) [inline gluten-ut/spark35/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:129]: Is there any issue to track this, if not please open one, thanks.

Comment (ayushi-agarwal) [reply gluten-ut/spark35/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:129]: Sure, I will open one, this is the PR where get_json_object support was added https://github.com/facebookincubator/velox/commit/20e467869b86d3c6d355cf52a550b94a10710479

**Comment by zhouyuan** [comment]
CC @PHILO-HE

File: gluten-ut/spark35/src/test/scala/org/apache/spark/sql/GlutenJsonFunctionsSuite.scala [hunk 2]
```diff
@@ -16,4 +16,88 @@
...
+    runTest("[\"a\",\"b\"]", "$[1]", "b")
+    runTest("[[\"a\",\"b\"]]", "$[0][1]", "b")
+
+    runTest("[1,2,3]", "[0]", "1")
+    // runTest("[1,2,3]", "$0", null) crashes in velox
+    runTest("[1,2,3]", "0", null)
+    runTest("[1,2,3]", "$.", null)
+
+    // runTest("[1,2,3]", "$", null) crashes in velox
+    // runTest("{\"a\":4}", "$", null) crashes in velox
```
Comment (PHILO-HE) [inline gluten-ut/spark35/src/test/scala/org/apache/spark/sql/GlutenJsonFunctionsSuite.scala:97]: I just created one issue to track the above two cases: https://github.com/apache/incubator-gluten/issues/5504.

According to Spark code, I note input json is the expected result for these two tests. Please keep them unchanged (still commented).

File: gluten-ut/spark35/src/test/scala/org/apache/spark/sql/GlutenJsonFunctionsSuite.scala [hunk 3]
```diff
@@ -16,4 +16,88 @@
...
+      ",\"bicycle\":{\"price\":19.95,\"color\":\"red\"}}" +
+      ",\"email\":\"amy@only_for_json_udf_test.net\"" +
+      ",\"owner\":\"amy\",\"zip code\":\"94025\",\"fb:testid\":\"1234\"}"
+
+    // Basic test
+    runTest(json, "$.owner", "amy")
+    runTest(json, "$.store.bicycle", "{\"price\":19.95,\"color\":\"red\"}")
+    runTest(json, "$.store.book", book)
+    runTest(json, "$.store.book[0]", book0)
+    runTest(json, "$.store.book[*]", null) // not supported in velox
```
[the remark this answers is not in the document]
Comment (ayushi-agarwal) [reply gluten-ut/spark35/src/test/scala/org/apache/spark/sql/GlutenJsonFunctionsSuite.scala]: Added, thanks for the suggestion @PHILO-HE

File: gluten-ut/spark35/src/test/scala/org/apache/spark/sql/GlutenJsonFunctionsSuite.scala (hunk 2 above)
Comment (ayushi-agarwal) [reply gluten-ut/spark35/src/test/scala/org/apache/spark/sql/GlutenJsonFunctionsSuite.scala]: Made the change

File: docs/velox-backend-limitations.md [hunk 4]
```diff
@@ -47,6 +47,8 @@ In certain cases, Gluten result may be different from Vanilla spark.
 #### JSON functions
 Velox only supports double quotes surrounded strings, not single quotes, in JSON data. If single quotes are used, gluten will produce incorrect result.
 
+Velox doesn't support [*] in path when get_json_object function is called and returns null instead
```
Comment (PHILO-HE) [inline docs/velox-backend-limitations.md:50]: Nit: please add "." at the end of statement.

**Review by PHILO-HE — APPROVED** [review]
Only one nit.

File: docs/velox-backend-limitations.md (hunk 4 above)
Comment (ayushi-agarwal) [reply docs/velox-backend-limitations.md]: Done

**Comment by ayushi-agarwal** [comment]
> Only one nit.

I have addressed it, thanks.

**Pull request merged**